### PR TITLE
Copy level & ascension data from GO to SRO

### PR DESCRIPTION
**Pull request by tiffanynwyeung**
## Describe your changes

**tl;dr** copy ascension/level data from GO to SRO, edited SR files calling GO method instead of SR method, added yarn cache packages for mac

The existing code in `CharacterData` is using `validateLevelAsc` in SRO from `libs/gi/util/` when we have the base function in `libs/sr/util/`. I copied over the extra functions and consts used in the GO version to the SRO version, and edited any files in SR that was calling the GO version instead. 

From what I can tell, HSR has all weapons using the same levels/ascensions regardless of rarity, so the "low" versions from GO did not seem necessary to copy over. Please let me know if I need to fix this. 

## Issue or discord link

https://discord.com/channels/785153694478893126/1183809278209970217/1212381167551127612

## Testing/validation

I clicked through the Character level boxes manually on the dev build to make sure the numbers checked out

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [ ] I have commented my code in hard-to understand areas.
- [ ] I have made corresponding changes to README or wiki.
- [ ] For front-end changes, I have updated the corresponding English translations.
- [ x ] I have run `yarn run mini-ci` locally to validate format and lint.
- [ ] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

**Review by nguyentvan7 — APPROVED** [review]
Looks good, thanks!
